plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java

```diff
@@ -184,16 +186,40 @@ rnd_nchar char(1000) NOT NULL,
     @Test
     void testSelectLimit()
     {
-        @Language("SQL")
-        String tableQuery = "CREATE TABLE faker.default.single_column (rnd_bigint bigint NOT NULL)";
-        assertUpdate(tableQuery);
+        assertUpdate("CREATE TABLE faker.default.single_column (rnd_bigint bigint NOT NULL)");
+
+        assertQuery("SELECT count(rnd_bigint) FROM (SELECT rnd_bigint FROM single_column LIMIT 5) a",
+                "VALUES (5)");
+
+        assertQuery("""
+                    SELECT count(rnd_bigint)
+                    FROM (SELECT rnd_bigint FROM single_column LIMIT %d) a""".formatted(2 * MAX_ROWS_PER_SPLIT),
+                "VALUES (%d)".formatted(2 * MAX_ROWS_PER_SPLIT));
 
+        assertQuery("SELECT count(distinct rnd_bigint) FROM single_column LIMIT 5",
+                "VALUES (1000)");
+
+        assertQuery("""
+                    SELECT count(rnd_bigint)
+                    FROM (SELECT rnd_bigint FROM single_column LIMIT %d) a""".formatted(MAX_ROWS_PER_SPLIT),
+                "VALUES (%d)".formatted(MAX_ROWS_PER_SPLIT));
+
+        // generating data should be deterministic
         @Language("SQL")
```

Comment:
This annotation looks redundant after this change because `formatted`

Reply:
Removed. Note, the IDE now highlights the usages of this variable.